### PR TITLE
[jmx][supervisor] JMXFetch standalone

### PR DESCRIPTION
**Pull request by yannmh**
Rebase #1364

### CODE REVIEW

File: jmxfetch.py [hunk 1]
```diff
@@ -6,9 +6,14 @@
 import subprocess
 import tempfile
 import time
+import sys
+import optparse
 
 # datadog
+from daemon import Daemon
```
Comment (remh) [inline jmxfetch.py]: You don't need this import anymore i think

File: jmxfetch.py [hunk 2]
```diff
@@ -238,153 +306,44 @@ def is_jmx_check(cls, check_config, check_name, checks_list):
...
-                '--status_location', r"%s" % path_to_status_file, # Path to the status file to write
-                command, # Name of the command
-            ]
-
-            subprocess_args.insert(4, '--check')
-            for check in jmx_checks:
-                subprocess_args.insert(5, check)
-
-            # Specify a maximum memory allocation pool for the JVM
-            if "Xmx" not in java_run_opts and "XX:MaxHeapSize" not in java_run_opts:
```
Comment (remh) [inline jmxfetch.py]: Is it expected that you are removing these lines ? i don't see them being readded.

File: jmxfetch.py [hunk 3]
```diff
@@ -39,93 +44,86 @@
...
 LINK_TO_DOC = "See http://docs.datadoghq.com/integrations/java/ for more information"
 
-class InvalidJMXConfiguration(Exception): pass
+
+class InvalidJMXConfiguration(Exception):
+    pass
+
 
 class JMXFetch(object):
+    """Daemon around JMXFetch process"""
```
Comment (remh) [inline jmxfetch.py]: This comment is not up to date i think

**Comment by yannmh** [comment]
Thanks @remh. It should be ready to merge now.

**Comment by remh** [comment]
Looks like you didn't add the entry in the other supervisors (source install and the package one)
https://github.com/DataDog/dd-agent/blob/yann/rebase-jmx-standalone/packaging/supervisor.conf
https://github.com/DataDog/dd-agent/blob/yann/rebase-jmx-standalone/packaging/datadog-agent/source/supervisord.conf

Also, did you check that supervisor was properly handling the case where jmxfetch shouldn't run? to make sure that supervisor won't try to restart it.

**Comment by yannmh** [comment]
Adding it.

Yes I did. When no JMX integration configuration file is found, JMXFetch will gracefully return a 0 exit status code and the Supervisor will not try to restart it.

File: packaging/supervisor.conf [hunk 4]
```diff
@@ -47,5 +47,12 @@ startretries=3
 priority=998
 user=dd-agent
 
+[program:jmxfetch]
+command=python jmxfetch.py
```
Comment (remh) [inline packaging/supervisor.conf]: This is not the right command for this supervisor.

File: packaging/supervisor.conf [hunk 5]
```diff
@@ -47,5 +47,12 @@ startretries=3
...
+stdout_logfile=jmxfetch.log
```
Comment (remh) [inline packaging/supervisor.conf]: For this supervisor, there shouldn't be any stdout_logfile

File: packaging/datadog-agent/source/supervisord.conf [hunk 6]
```diff
@@ -40,5 +40,12 @@ redirect_stderr=true
 priority=998
 startsecs=3
 
+[program:jmxfetch]
+command=python jmxfetch.py
+stdout_logfile=jmxfetch.log
```
Comment (remh) [inline packaging/datadog-agent/source/supervisord.conf]: This is not the right path for the log. It should be supervisord/logs/jmxfetch.log

**Comment by remh** [comment]
:+1:

**Pull request merged**